### PR TITLE
feat: add convenience targets for working on docs

### PR DESCRIPTION
**Pull request by krancour**
@fykaa let me know if these are the sort of things that would have saved you some trouble up front and whether you think you could documents these targets as part of #2491.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -358,3 +360,18 @@ start-controller-local:
...
+# Convenience targets for building and running the documentation site natively #
+# or in a container.                                                           #
+################################################################################
+
+.PHONY: hack-docs
+hack-docs: hack-build-dev-tools
+	$(CONTAINER_RUNTIME) run $(DOCKER_OPTS) -p $(DOCS_PORT):$(DOCS_PORT) kargo:dev-tools make docs
+
+.PHONY: docs
+docs:
```
Comment (hiddeco) [inline Makefile:376]:
```suggestion
serve-docs:
```

maybe?

Comment (krancour) [reply Makefile:376]: I can get down with that.

Comment (krancour) [reply Makefile:376]: After thinking about it, I do slightly favor `serve-docs` / `hack-serve-docs` despite being a bit more verbose. There are other doc-related targets -- and for different kinds of docs, no less. For instance, there is `codegen-docs`, which deals with generating chart docs from the metadata in the Helm chart's `values.yaml` file. Naming things is hard. Humans, by nature, are easily confused... so I am often willing to sacrifice brevity for clarity.

Comment (krancour) [reply Makefile]: 3b999b7849826600f4f88e4fad8521987fa3f69e updates the names of the targets as discussed above.

**Review by hiddeco — APPROVED** [review]

**Pull request merged**